src/ListGuesser.js

```diff
@@ -24,59 +25,62 @@ const displayOverrideCode = (resourceSchema, fields) => {
   console.info(code);
 };
 
-export class ListGuesserComponent extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      orderParameters: this.props.resourceSchema
-        ? getOrderParametersFromResourceSchema(this.props.resourceSchema)
-        : [],
-    };
-  }
-
-  componentDidMount() {
-    if (this.state.orderParameters.length) {
-      return;
-    }
+export const ListGuesserComponent = ({
+  children,
+  resourceSchema,
+  fields,
+  resourceSchemaAnalyzer,
+  ...props
+}) => {
+  const [orderParameters, setOrderParameters] = useState([]);
 
-    this.props.resourceSchema.getParameters().then(() => {
-      this.setState({
-        orderParameters: getOrderParametersFromResourceSchema(
-          this.props.resourceSchema,
-        ),
-      });
-    });
-  }
+  useEffect(() => {
+    if (resourceSchema) {
+      const resolvedOrderParameters = resourceSchemaAnalyzer.getOrderParametersFromResourceSchema(
+        resourceSchema,
+      );
 
-  render() {
-    const {resourceSchema, fields, ...props} = this.props;
+      setOrderParameters(resolvedOrderParameters);
```

Comment:
I don't understand why you setState here even though `resolvedOrderParameters` might be empty, and you'll have to call the `resourceSchema` just next line to set it again.

Reply:
It seems it's because sometimes the order parameters are already in the parsed schema, and sometimes it's not and a supplementary call to the API is needed in order to retrieve them. I'm not sure why but maybe @dunglas or @mauchede can answer?

Reply:
Actually, Hydra doesn't support (yet) to expose available filters in `docs.jsonld`, this information is only available in the metadata of the collection, so a call to the 1st page is needed to retrieve these parameters.